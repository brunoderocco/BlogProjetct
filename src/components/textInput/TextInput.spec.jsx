
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { TextInput } from '.';

describe('<TextInput />', () => {

  it('should have a value of shearchValue ', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} searchFor='Testando' />);

    const input = screen.getByPlaceholderText(/type your search/i);
    expect(input).toBeInTheDocument();

    expect(input.value).toBe('Testando');

  });

  it('should call handle Change on each key pressed', () => {
    const fn = jest.fn();
    render(<TextInput handleChange={fn} />);

    const input = screen.getByPlaceholderText(/type your search/i);

    const value = 'o valor';
    userEvent.type(input, value);

    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match snapshop', () => {
    const fn = jest.fn();
    const { container } = render(<TextInput handleChange={fn} />);
    expect(container.firstChild).toMatchSnapshot();
  });
}); 