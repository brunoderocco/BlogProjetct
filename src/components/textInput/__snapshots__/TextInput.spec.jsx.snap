// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshop 1`] = `
<div>
  <input
    class="input"
    placeholder="type your search"
    type="search"
    value=""
  />
</div>
`;
